### DISCUSSION TITLE
IP limitation access

### DISCUSSION THREAD

**Discussion by Ramel** [mar10/wsgidav#275] [Q&A]
Hello,
Is there a way to limit access to the Wsgidav from a specific IP, or list of IP?
Without using a proxy for example.
Thanks

**Comment by Ramel** [comment]
Asked on Cheroot's chat:
@Ramel
[nov. 14 12:34](https://gitter.im/cherrypy/cherrypy?at=637227e2bcdb0060b81f8e08)
Hello, simple question here: Is there a way to restrict by IP the access to Cheroot in some parameters?
@webknjaz
[nov. 15 12:53](https://gitter.im/cherrypy/cherrypy?at=63737d9d2cb65702402aecf5)
Cheroot is quite low-level. You'd have to do this in WSGI handlers. FWIW this is something that is better solved on a higher abstraction layer, in a framework, like CherryPy.
@Ramel
[nov. 15 16:30](https://gitter.im/cherrypy/cherrypy?at=6373b082473cf96648e8c22f)
Ok, but as I'm using WsgiDAV, I've not find, yet, a parameter that can do the trick. Using " domain_controller: wsgidav.dc.pam_dc.PAMDomainController" in wsgidav conf file
@webknjaz
nov. 15 17:38m
Ask on their tracker maybe?

**Comment by mar10** [comment]
WsgiDAV does not provide IP filtering, whitelisting etc.
I would think this is a feature that would be provided by a proxy or the WSGI container. Cheroot is only one of many options.

However it would be possible to write a middleware and add it to the [WsgiDAV stack](https://wsgidav.readthedocs.io/en/latest/reference_guide_architecture.html#wsgi-application-stack).
The [CORS middleware](https://github.com/mar10/wsgidav/blob/master/wsgidav/mw/cors.py) may serve as an example.
